manifests/charts/gateway/templates/hpa.yaml

```diff
@@ -1,5 +1,6 @@
 {{- if .Values.autoscaling.enabled }}
-apiVersion: autoscaling/v2beta2
+{{- if (semverCompare ">=1.23-0" .Capabilities.KubeVersion.GitVersion)}}
```

Comment:
nit: since the only change is API version should we just have the `if` around the `apiVersion`?

Reply:
done